ctapipe/io/simteleventsource.py

```diff
@@ -0,0 +1,162 @@
+from ctapipe.io.eventsource import EventSource
+from ctapipe.io.hessioeventsource import HESSIOEventSource
+from ctapipe.io.containers import DataContainer
+from astropy import units as u
+from astropy.coordinates import Angle
+from astropy.time import Time
+from ctapipe.instrument import TelescopeDescription, SubarrayDescription
+
+from eventio.simtel.simtelfile import SimTelFile
+
+__all__ = ['SimTelEventSource']
+
+import eventio
+
+class SimTelEventSource(EventSource):
+
+    def __init__(self, config=None, tool=None, **kwargs):
+        super().__init__(config=config, tool=tool, **kwargs)
+        self.metadata['is_simulation'] = True
+        self.file_ = SimTelFile(self.input_url)
+
+        self._subarray_info = self.prepare_subarray_info()
+
+    def prepare_subarray_info(self):
+        """
+        constructs a SubarrayDescription object from the info in an
+        EventIO/HESSSIO file
+
+        Parameters
+        ----------
+        file: HessioFile
+            The open pyhessio file
+
+        Returns
+        -------
+        SubarrayDescription :
+            instrumental information
+        """
+
+        subarray = SubarrayDescription("MonteCarloArray")
+
+        for tel_id, cam_settings in self.file_.cam_settings.items():
+            tel = TelescopeDescription.guess(
+                cam_settings['pixel_x'] * u.m,
+                cam_settings['pixel_y'] * u.m,
+                equivalent_focal_length=cam_settings['focal_length'] * u.m
+            )
+            tel.optics.mirror_area = cam_settings['mirror_area'] * u.m ** 2
+            tel.optics.num_mirror_tiles = cam_settings['mirror_area']
+            subarray.tels[tel_id] = tel
+            H = self.file_.header
+            subarray.positions[tel_id] = H['tel_pos'][H['tel_id'] - 1] * u.m
+
+        return subarray
+
+    @staticmethod
+    def is_compatible(file_path):
+        # can be copied here verbatim if HESSIOEventSource should be removed
+        return HESSIOEventSource.is_compatible(file_path)
+
+    def _generator(self):
+
+
+        data = DataContainer()
+        data.meta['origin'] = "eventio"
+        data.meta['input_url'] = self.input_url
+        data.meta['max_events'] = self.max_events
+
+        for counter, (shower, event) in enumerate(self.file_):
+            # next lines are just for debugging
+            self._event = event
+            self._shower = shower
+            # I did not exactly understand which event_id is used
+            # here .. tests agains the pyhessio implementation will
+            # hopefully show this
+            event_id = counter
```

Comment:
I think this should be the `event_id` of the last `MCEvent`, so `shower * 100 + reuse`, that would make events uniquely identifiable together with the `run_id`